File: src/main/java/org/cip4/xjdf/json/openapi/model/Operation.java
package org.cip4.xjdf.json.openapi.model;

import lombok.Builder;
import lombok.Data;

import java.util.List;
import java.util.Map;

@Data
@Builder
public class Operation {
    private final List<String> tags;
    private final RequestBody requestBody;
    private final Map<String, Response> responses;
    private final Map<String, Map<String, PathItem>> callbacks;
}

File: src/main/kotlin/org/cip4/xjdf/openapi/model/Operation.kt
package org.cip4.xjdf.openapi.model

import kotlinx.serialization.Serializable

@Serializable
data class Operation (
    val tags: List<String>? = null,
    val requestBody: RequestBody? = null,
    val responses: Map<String, Response>? = null,
    val callbacks: Map<String, Map<String, PathItem>>? = null,
)
